### PR TITLE
feat(shared-lessons-list)[skp-265]: Creates lessons list

### PR DESCRIPTION
**Pull request by Gardemarin**

### CODE REVIEW

File: libs/shared/lessons/src/lib/shared-lessons.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { List } from '@material-ui/core';
+import './shared-lessons.scss';
+import LessonItem, { Action, Lesson } from './lesson-item/lesson-item';
+
+/* eslint-disable-next-line */
```
Comment (doncevas) [inline libs/shared/lessons/src/lib/shared-lessons.tsx:6]: remove comment

File: libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        <address className="author lesson__text">
+          By
+          <a rel="author" href="#">
+            {props.lesson.creator.name}
+          </a>
+        </address>
+        <time className=" lesson__text" dateTime={props.lesson.datime}>
+          {new Date(props.lesson.datime).toDateString()}
+        </time>
+        <span className=" lesson__text">2h30min</span>
```
Comment (doncevas) [inline libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx:67]: what is '2h30min'?

File: libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        </Link>
+
+        <address className="author lesson__text">
+          By
+          <a rel="author" href="#">
+            {props.lesson.creator.name}
+          </a>
+        </address>
+        <time className=" lesson__text" dateTime={props.lesson.datime}>
+          {new Date(props.lesson.datime).toDateString()}
```
Comment (doncevas) [inline libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx:65]: wee using moment.js for better time and date handling

File: libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  const open = Boolean(anchorEl);
+
+  const preventDefault = (event: React.SyntheticEvent) => {
+    event.preventDefault();
+  };
+
+  return (
+    <ListItem className="lesson" alignItems="flex-start">
+      <div className="lesson__details">
+        <Link className="lesson__title" href="#" onClick={preventDefault}>
```
Comment (doncevas) [inline libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx:54]: it is link or event function ?  same below

File: libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const open = Boolean(anchorEl);
+
+  const preventDefault = (event: React.SyntheticEvent) => {
+    event.preventDefault();
```
Comment (doncevas) [inline libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx:48]: what the target of this function?

File: libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const open = Boolean(anchorEl);
```
Comment (doncevas) [inline libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx:45]: what for this variable? maybe you can use state directly ?

File: libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(
+    null
+  );
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
```
Comment (doncevas) [inline libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx:42]: this handler handling only one function, maybe you can use this state function directly ?  same above handleClick function

File: libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+  description: string;
+}
+
+export interface Action {
+  title: string;
+  action(lesson: Lesson): void;
+}
+
+export const LessonItem = (props: { lesson: Lesson; actions: Action[] }) => {
+  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(
```
Comment (doncevas) [inline libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx:33]: import useState on top

File: libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+    name: string;
+    imageUrl?: string;
+  };
+  imageUrl: string;
+  description: string;
+}
+
+export interface Action {
+  title: string;
+  action(lesson: Lesson): void;
```
Comment (doncevas) [inline libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx:29]: Action have action ? it is function or field? if it is function it should look like this: action: (lesson: Lesson) => void

**Review by doncevas — CHANGES_REQUESTED** [review]

File: libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx (hunk 2 above)
Comment (Gardemarin) [reply libs/shared/lessons/src/lib/lesson-item/lesson-item.tsx:67]: Length of the lesson. But now back-end doesn't have a lesson end date, so I left how it is, or we can replace to something else. Not sure what ...

**Review by doncevas — APPROVED** [review]

**Pull request merged**